### ISSUE TITLE
Failover not working (no internet) RT-AC5300

### ISSUE THREAD

**Issue by DukenStein1099** [Ranger802004/asusmerlin#49]
First post. I still can't get the failover to work (internet) and I'm not sure if it's my config of the script, router or my failover device. If someone could point me in the right direction it would be much appreciated. I am sure I'm doing something stupid because I'm a noob.

Router - RT-AC5300 fw 386.14
Dual Wan enabled, USB Modem enabled
Orbic RC400L, Verzion Jetpack, connected to asus usb, tethering enabled on jetpack
Installed Ranger's latest script (via amtm). default settings.

When I pull the primary internet connection I can see that the script correctly fails over "I think" from Wan0 to Wan1 (jetpack) but I can't get to the internet. The jetpack works fine if I connect a device to it using wifi so I know the jetpack itself works. It appears the jetpack is recognized in the router and the script fails over to it but why no internet? Is there something additionally I need to be configuring in the script or in the router?

The router has dhcp enabled using 192.168.1.1 The jetpack also has dhcp enabled (should I disable it?) but since it was using the same ip subnet as the rotuer I changed it to use 192.168.10.1. Is that ok for it to be on a different subnet? I'm thinking that my problem has to do with this dhcp config between the two devices but I want to make sure I'm not missing something.

Any thoughts or advice much appreciated..

**Comment by Ranger802004** [comment]
Do you have debug logs from WAN Failover available? I would recreate this scenario capturing debug logs.  Make sure debug logging is enabled under System Logs in the GUI and then run WAN Failover with capture option from the menu and then recreate your test scenario.  When done end the capture logs function and collect the file it displays afterwards.

**Comment by DukenStein1099** [comment]
Ok I will do that when I’m home later and post back

**Comment by DukenStein1099** [comment]
Finally back from traveling and able to test. I hope I did the logging correctly (noob) and I'm attaching file 
[Nov  3 073114 wan-failover Debug -.txt](https://github.com/user-attachments/files/17610488/Nov.3.073114.wan-failover.Debug.-.txt)
below. Also, is the "allow failback" supposed to be checked in the asus gui? I was confused in the installation instructions. I tried it both ways but figured it wouldn't affect the failover results either way. Any thoughts/suggestions appreciated. Thanks.

**Comment by Ranger802004** [comment]
WAN Failover is not taking over for your failover functionality because you haven't disabled all of the ASUS factory failover options as documented in the readme.

**Nov  3 07:31:17 wan-failover: WAN Status - ASUS Factory Watchdog: Enabled**
Nov  3 07:31:17 wan-failover: Debug - Function: wandisabled
**Nov  3 07:31:17 wan-failover: WAN Failover Disabled - WAN Failover is currently disabled.  ***Review Logs*****
Nov  3 07:31:18 wan-failover: WAN Failover Disabled - USB Device for wan1 is in Ready State but in Cold Standby
Nov  3 07:31:18 wan-failover: Debug - Function: cleanup
Nov  3 07:31:18 wan-failover: Debug - Setting parameters for wan0
Nov  3 07:31:18 wan-failover: Debug - Setting parameters for wan0
Nov  3 07:31:18 wan-failover: Debug - Checking wan0 for IP Rule to 8.8.8.8
Nov  3 07:31:18 wan-failover: Debug - Checking wan0 for Default Route in 100
Nov  3 07:31:18 wan-failover: Debug - Setting parameters for wan1
Nov  3 07:31:18 wan-failover: Debug - Setting parameters for wan1
Nov  3 07:31:19 wan-failover: Debug - Checking wan1 for IP Rule to 8.8.4.4
Nov  3 07:31:19 wan-failover: Debug - Checking wan1 for Default Route in 200
Nov  3 07:31:19 wan-failover: Debug - Checking for Lock File: /var/lock/wan-failover.lock
Nov  3 07:31:19 wan-failover: Cleanup - Deleting /var/lock/wan-failover.lock
Nov  3 07:31:19 wan-failover: Cleanup - Deleted /var/lock/wan-failover.lock

**Comment by DukenStein1099** [comment]
I figured I was doing something dumb. I will go back and review the readme and update results Thanks!

**Comment by DukenStein1099** [comment]
I'm sorry if I am asking a really obvious question but where do I "disable factory failover"? I see this in the readme file:
Requirements:
- ASUS Merlin Firmware v386.5 or higher **_(yes)_**
- JFFS custom scripts and configs Enabled **_(yes)_**
- Dual WAN Enabled **_(yes)_**
- ASUS Factory Failover Disabled (Network Monitoring Options, Allow Failback Option under WAN > Dual WAN) **_(???)_**

I'm not sure what the last one is. I don't see anywhere for "Network Monitoring Options". 

Under the Wan/Dual Wan tab I have Dual Wan enabled and then there is a choice for Dual Wan Mode (Failover or Load Balancing, I have Failover). There is then a checkbox for whether you want to Allow Failback. I have tried both checked and unchecked without success. I have also tried both checking/unchecking DNS query/Ping, also without success. Is there somewhere else where I should be "disabling the factory failover". Attaching pic. 
![asus dual wan config](https://github.com/user-attachments/assets/27598f2c-4c7b-4b5a-b2a3-06e75bf9dcde)

**Comment by Ranger802004** [comment]
> * ASUS Factory Failover Disabled (Network Monitoring Options, Allow Failback Option under WAN > Dual WAN) **_(???)_**

ASUS Factory Failover Disabled (**Network Monitoring Options**, **Allow Failback** Option under WAN > Dual WAN) (???)

<img width="606" alt="image" src="https://github.com/user-attachments/assets/24b6eaac-843a-4a54-9308-10a8f10d386c">

**Comment by DukenStein1099** [comment]
ok thanks. I will try that later this afternoon. I did try it this morning and I wasn't able to access the internet but maybe now it's a different problem and it is failing over correctly but it's something else. I will run another capture log later and see what it says.

**Comment by Ranger802004** [comment]
> ok thanks. I will try that later this afternoon. I did try it this morning and I wasn't able to access the internet but maybe now it's a different problem and it is failing over correctly but it's something else. I will run another capture log later and see what it says.

Any updates?

**Comment by DukenStein1099** [comment]
yes sorry was just able to do it. Here's the latest capture with the settings unchecked pics and file attached.
![IMG_0503](https://github.com/user-attachments/assets/6613ba50-19b1-4fdf-a727-9ea0868d64e0)
![IMG_0502](https://github.com/user-attachments/assets/0f8e205e-cb0b-4c50-9a2a-c28e8c7572a6)
[11072024.wan-failover.Debug.-.txt](https://github.com/user-attachments/files/17669967/11072024.wan-failover.Debug.-.txt)

**Comment by Ranger802004** [comment]
How is your Primary WAN connected?

**Comment by DukenStein1099** [comment]
The Primary Wan is connected via ethernet to the wan port on the router. For the test I initially left it plugged in, started capture in the script, then unplugged the the primary to simulate a primary failure. You can see from my pics that the router shows the primary wan as unplugged/disconnected and it shows the secondary as connected. I'm not sure what the capture is saying because I'm unsure how to interpret it. From the asus router gui it appears to be failing over but not grabbing an ip from the jetpack.

Do I have the router settings configured properly now?

**Comment by Ranger802004** [comment]
Is it possible for you to test without physically unplugging your WAN0? Maybe there is a connection to your Primary ISP you can disrupt to test? When you unplug it, your router is still trying to switch to the WAN1 for some reason and causing some issues with WAN Failover switching to WAN1.

**Comment by DukenStein1099** [comment]
sure i will do it right now and post back

**Comment by DukenStein1099** [comment]
ok I pulled the power on my fios nic and left the physical ethernet connection plugged into the asus router. That brought down the internet connection but left all physical cables plugged into the router. As you can see from the pic it says "network cable unplugged" but nothing is unplugged from the router, just the main fios power cable. The router gui says it has switch to wan 1 but it still doesn't show any wan ip/internet. I am also attaching the capture log. Thanks.
![IMG_0504](https://github.com/user-attachments/assets/8ee8544b-1be9-40e4-853c-36c92e83c0de)
[11082024.wan-failover.Debug.-.txt](https://github.com/user-attachments/files/17684253/11082024.wan-failover.Debug.-.txt)

**Comment by DukenStein1099** [comment]
are you able to tell from the logs whether the script appears to be working? Is it failing over? Trying to narrow down whether it's the configuration of the jetpack pulling an ip or if I'm not even getting that far. Thanks

**Comment by DukenStein1099** [comment]
yes fios is primary. So does that mean the script is working? Failing over to secondary? I'm not sure from the logs, I still don't understand it all. If the script is failing over correctly, then I'll start working on figuring out why the jetpack isn't pulling an ip. I just want to double check whether the script is failing over correctly.

**Comment by Ranger802004** [comment]
It looks like it's getting an IP but the problem is your WAN0 is showing disconnected.  Do you have a faulty ethernet cable plugged up for WAN0 by any chance? I have seen this before.

**Comment by DukenStein1099** [comment]
I will check the cables and try bringing the internet connection down using a different method but wouldn't the failover work the same even if a cable was unplugged since the internet is still "down"?

**Comment by Ranger802004** [comment]
It shouldn't show as unplugged unless it physically sees the cable disconnect from the router.

**Comment by DukenStein1099** [comment]
ok I am going to try and figure out bouncing the internet later without it saying cable unplugged

**Comment by Ranger802004** [comment]
Any updates on your issue?

**Comment by Ranger802004** [comment]
Following up on this, have you had any success?

**Comment by DukenStein1099** [comment]
Hi Ranger thanks for checking back. I just got back from traveling so didn't have a chance to check until today. I have tried every way I can think of to bounce the internet connection without the router displaying "network cable unplugged" but all ways result in that message. Unplugging the cable, unplugging the connected switch, killing power on the FIOS ONT, all have the same result, which makes sense since all will kill the ethernet connection. Only other thing I can think of is to change the gateway/dns/ip in the FIOS router but that wouldn't make sense to do because that isn't a real life scenario. Most likely is for one of the other events to happen which would result in that message.

Is there any way to make this work even though it says "cable unplugged"?

**Comment by DukenStein1099** [comment]
I thought of something that I just tried but didn't work. I plugged the router wan ethernet into an empty switch so that it would still show the cable plugged in but would not be able to get internet from the fios gateway, which should simulate an internet down scenario. The router then showed Primary Wan as "your isp's dhcp does not function properly". It also shows the router switched to Secondary Wan. I could not however get on the internet and the failover did not appear to be working. It was also interesting that the capture log seemed pretty empty like nothing was happening or logging. Posting below

Nov 28 11:51:41 wan-failover: Debug - Function: monitor
Nov 28 11:51:41 wan-failover: Debug - Function: setvariables
Nov 28 11:51:41 wan-failover: Debug - Reading /jffs/configs/wan-failover.conf
Nov 28 11:51:41 wan-failover: Debug - Checking if System Log is located at /tmp/syslog.log and isn't a blank file
Nov 28 11:51:41 wan-failover: Debug - System Log is located at /tmp/syslog.log
Nov 28 11:52:32 wan-failover: WAN Failover Disabled - Failover Mode: wan1 is Primary
Nov 28 11:52:33 wan-failover: WAN Failover Disabled - wan0 does not have a valid IP Address: 0.0.0.0
Nov 28 11:52:33 wan-failover: WAN Failover Disabled - wan0 does not have a valid Gateway IP Address: 0.0.0.0
Nov 28 11:52:41 syslog: wlceventd_proc_event(530): eth1: Auth 4C:65:A8:33:9B:68, status: Successful (0), rssi:0
Nov 28 11:52:41 syslog: wlceventd_proc_event(559): eth1: Assoc 4C:65:A8:33:9B:68, status: Successful (0), rssi:0
Nov 28 11:52:42 dnsmasq-dhcp[447]: DHCPREQUEST(br0) 192.168.1.112 4c:65:a8:33:9b:68 
Nov 28 11:52:42 dnsmasq-dhcp[447]: DHCPACK(br0) 192.168.1.112 4c:65:a8:33:9b:68 Roost-device-9B68
Nov 28 11:52:43 syslog: wlceventd_proc_event(511): eth1: Disassoc 4C:65:A8:33:9B:68, status: 0, reason: Unspecified reason (1), rssi:0
Nov 28 11:53:00 kernel: 00:40:7F:76:2B:B2 not mesh client, can't delete it
Nov 28 11:53:00 kernel: 00:40:7F:76:2B:C2 not mesh client, can't delete it
Nov 28 11:54:09 syslog: wlceventd_proc_event(511): eth3: Disassoc 5A:A8:5E:9C:32:A0, status: 0, reason: Disassociated because sending station is leaving (or has left) BSS (8), rssi:0
Nov 28 11:55:09 dnsmasq-dhcp[447]: DHCPREQUEST(br0) 192.168.1.106 00:21:bd:e3:db:62 
Nov 28 11:55:09 dnsmasq-dhcp[447]: DHCPACK(br0) 192.168.1.106 00:21:bd:e3:db:62 Wii
Nov 28 11:56:00 kernel: 5C:B0:66:AC:23:80 not mesh client, can't delete it
Nov 28 11:56:02 kernel: E0:94:67:B7:A7:00 not mesh client, can't delete it

![router](https://github.com/user-attachments/assets/225594de-b1e8-463b-8ffc-43e4af3e84eb)

**Comment by Ranger802004** [comment]
The only thing I'm really seeing is that your router is failing to add required routes on the specific routing tables for each WAN interface.  Do you know why these might be failing? Are you sure your router is in a fully healthy state? These would make WAN Failover think both of the WAN interfaces are down.

> Nov 28 12:20:57 wan-failover: Check Routing Table - ***Error*** Unable to add default route to Target IP: 8.8.8.8 for wan0 Routing Table via 96.225.76.1 dev eth0
> Nov 28 12:20:58 wan-failover: Check Routing Table - ***Error*** Unable to add default route to Target IP: 8.8.4.4 for wan1 Routing Table via 192.168.10.1 dev usb0

**Comment by DukenStein1099** [comment]
Hey Ranger somehow I missed this post. The router and internet seemed to be working perfectly and configured correctly other than the failover not working. Everything I checked with the router appears solid and all traffic works seamlessly. I'm wondering if this has anything to do with the subnet the jetpack is on. Does it need to be on the same subnet as the router? Is that why it's failing or is it not even getting that far?

**Comment by Ranger802004** [comment]
> Hey Ranger somehow I missed this post. The router and internet seemed to be working perfectly and configured correctly other than the failover not working. Everything I checked with the router appears solid and all traffic works seamlessly. I'm wondering if this has anything to do with the subnet the jetpack is on. Does it need to be on the same subnet as the router? Is that why it's failing or is it not even getting that far?

The behavior you're experiencing I've seen when the router sees a physical link failure and still tries to failover to the other WAN using the factory method.  Double check you have disabled all of the Failback / Monitoring options in the GUI.

**Comment by Ranger802004** [comment]
Following up with you on your issue, any updates?

**Comment by DukenStein1099** [comment]
Hi Ranger, no progress. I’ve hit a wall. I’m not sure what else to try at this point. Everything seems to be configured properly and each device functions well on its own. I just don’t understand where the breakdown is. 

I am considering going with a dual wan router (that works better) inside the lan as a bridge. Let me know if you have any other thoughts. Thanks

**Comment by Ranger802004** [comment]
I am investigating some things on the script side as well so if you can wait I may have some things for you to test.

**Comment by DukenStein1099** [comment]
That’s great. I’ll wait. Right now I am just using the jet pack tethered to a critical pc but ideally want it connected to the router. Let me know when you are ready to test.

**Comment by Ranger802004** [comment]
Please test on latest beta release (v2.1.3-beta1).

**Comment by Ranger802004** [comment]
Closing for no response.
